src/scf/self_consistent_field.jl

```diff
@@ -4,22 +4,24 @@ include("scf_callbacks.jl")
 # (unused in primal calculations)
 @kwdef struct ResponseOptions
     verbose = false
-    occupation_threshold = 1e-10
 end
 
 function default_n_bands(model)
     min_n_bands = div(model.n_electrons, filled_occupation(model), RoundUp)
     n_extra = model.temperature == 0 ? 0 : max(4, ceil(Int, 0.2 * min_n_bands))
     min_n_bands + n_extra
 end
+default_occupation_threshold() = 1e-10
```

Comment:
What if we are using single precision or other precisions? It feels there needs to be a floating-point type argument passed here.